c7n/resources/rds.py

```diff
@@ -1454,8 +1455,11 @@ def get_permissions(self):
 
     def process(self, configs, event=None):
         rds = self.manager.get_resource_manager('rds').resources()
-        self.used = {r.get('DBSubnetGroupName', r['DBInstanceIdentifier'])
-                     for r in rds}
+        self.used = set(jmespath.search('[].DBSubnetGroup.DBSubnetGroupName', rds))
+        self.used.union(
```

Comment:
union returns a new set.. you want update

Reply:
Thanks again @kapilt! Resolved in 6f0c0f7.